doc/tests-updates.md

```diff
@@ -0,0 +1,381 @@
+was received. In this case `response` is set to `None`.
+
+If received response doesn't match expected one, then an error message is generated:
+```
+Error on request-response processing!
+For request:
+---
+<Request>
+---
+Response was expected:
+---
+<Expected response>
+---
+But received response was:
+---
+<Received response>
+---
+```
+Error should also be generated if the connection was closed unexpectedly or
+connection wasn't closed as expected.
+
+
+If debug is enabled, write to file `debug_filename_%connid%` any activity on the
+connection with the timestamp: received bytes, sent bytes, and parsed messages.
+
+
+#### Mock Server
+
+Follows the same concept as mock client. Arguments:
+
+- Script. - filename with the request/response functions. The content of it
+is described later.
+
+- ID. - server id.
+
+- Interface:conns. - interfaces the client sockets will be listening and number
+of Tempesta's connections for it.
+
+- Split. - Boolean value, meaning that requests must be splited in multiple skbs.
+Don't know how to implement it yet.
+
+- Debug file. - Filename to write debug information.
+
+Server is stopped by `TERM` signal from the framework.
+
+Script contains queue of expected requests `server_data.requests[]`. If all
+requests was received, but a new request is received, then client is working in
+repeated mode, thus `conn_data.expected_request_num` must be reset to `0`.
+
+`def gen_response_func(server_data, conn_data, request)` function is used to
+give a response for the client. Returns `response` or `None`. The received
+request is compared with `server_data.requests[T-Req-Id]`
+but note, that `T-Client-Id`, `T-Conn-Id`, `T-Req-Id` values may be missed in the
+expected request.
+
+The generated response **must** contain headers copied from the request:
+`T-Client-Id`, `T-Conn-Id`, `T-Req-Id`. `T-Srv-Id` and `T-Srv-Conn-Id` must
+also be generated.
+
+If the server need to close the connection in response to request, it sets
+`conn_data.close` variable. The variable `conn_data.expect_close` can also be
+set if generated response is invalid and Tempesta will reset the connection.
+
+
+### Framework
+
+The framework now is defined not as list of actions but as HTTP flows on client
+and server. This allows to replace Tempesta by other HTTP ADC software and easily
+compare behavior between them.
+
+This also allows to create multiple tests from the same description:
+
+- Message integrity, Invalid message processing, Cache tests: single client,
+single client connection.
+
+- Scheduler, Workload, Request-response chain integrity tests: multiple clients
+from many source interfaces, multiple concurrent connections for each client.
+
+- Fuzzing tests: Workload tests, but every generated messages are split into
+different number of skbs.
+
+When reduced number of tests is executed, only first group of tests is executed,
+the full set is for first and second groups. All groups are executed in background
+stability tests.
```

Comment:
Make sense for generic case, but not so much sense if a developer changes low level logic w/o changing any features and he's curious about network operation. I propose to be able to set this in config as well as for particular tests. In this way you can run for example a particular TLS test with message integrity check and/or fuzzing (strictly speaking, skb splitting isn't a real fuzzing) at the same time some tests oriented on the low level logic have sense only with one of few options enabled.